src/lua/lua_script.rs

```diff
@@ -279,10 +280,12 @@ impl LuaScript {
         result.ok()
     }
 
-    pub fn set_prompt_content(&mut self, content: String) {
+    pub fn set_prompt_content(&mut self, content: String, pos: usize) {
         self.exec_lua(&mut || -> LuaResult<()> {
             self.state
                 .set_named_registry_value(PROMPT_CONTENT, content.clone())?;
+            self.state
+                .set_named_registry_value(PROMPT_CURSOR_INDEX, pos)?;
```

Comment:
We also need to update the `PROMPT_CURSOR_INDEX` registry value after processing command binds since those directly manipulate the buffer's pos. Right now it seems like the value quickly gets out of sync with the cursor position.

Here's a script I used to demonstrate the problem:
```lua
blight.bind("f5", function ()
    blight.output("moving cursor");
    prompt.set_cursor_pos(4)
end)
blight.bind("f6", function ()
    local pos = prompt.get_cursor_pos()
    blight.output("cursor pos: ".. tostring(pos));
end)
```

1. If you type some input and press `f6` the position at the end of the buffer is correctly reported :heavy_check_mark: 
2. If you press `f5` the cursor is moved to position 4 and the prompt redrawn :heavy_check_mark: 
3. If you press `f6` again, the position is still reported as the end of the buffer :x: 

I think you'll need to add a `luascript.rs` function to set the registry value, and then call that from `command.rs` in the `termion::event::Event::Key` match arm that I was mucking with in https://github.com/Blightmud/Blightmud/pull/738. I think we want to update the prompt cursor value in the place that emits `UserInputCursor` and the place that emits `UserInputBuffer`.

Reply:
I had a rough WIP branch with logic that accounts for the above if its helpful: https://github.com/Blightmud/Blightmud/commit/1354844e84054ea367cc61f5e2740c90c81570d2.

Reply:
Nice spot. I'll dig into it